python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -34,6 +42,47 @@
 from .valid_definitions import VALID_REPOSITORY_DATA_DICT_KEYS, RepositoryListDefinition
 
 
+def _find_env_vars(config_entry: Any) -> Set[str]:
+    """
+    Given a part of a config dictionary, return a set of environment variables that are used in
+    that part of the config.
+    """
+    if isinstance(config_entry, list):
+        return set().union(*[_find_env_vars(v) for v in config_entry])
+    elif not isinstance(config_entry, dict):
```

Comment:
worth noting here what the universe of possible types are when it is not a dict

also just a minor thing but do I think doing

```
if isinstance(config_entry, list):
    # do list-only things
    
if isinstance(config_entry, dict):    
    # do dict-only things
    
# fallback code
```

if marginally more obvious to the reader than the `elif not isinstance(config_entry, dict):` clause